### ISSUE TITLE
Attempt to add an already existing user results in server error 500 (v1.0.0)

### ISSUE THREAD

**Issue by KonstantinShemyak** [stevespringett/dependency-track#66]
Steps to reproduce:

1. Log in to Dependency Track (v1.0.0) as `admin`, go to `user management`
2. Add a username which already exists

Result: Server error 500. The meaningful stack trace line:

```
org.h2.jdbc.JdbcSQLException: Unique index or primary key violation: "CONSTRAINT_INDEX_4D ON PUBLIC.USERS(USERNAME)"; SQL statement:
insert into USERS (ID, CHECKVALID, ISLDAP, PASSWORD, ROLEID, USERNAME) values (null, ?, ?, ?, ?, ?) [23505-172]
```

**Comment by stevespringett** [comment]
I have verified this is no longer an issue with 3.0 however 3.0 doesn't provide any feedback that a user already exists.

**Comment by stevespringett** [comment]
Fixed in 3.0-dev branch. Error messages in the UI now use the exact same error messages that are sent from the REST API (and defined in the Swagger definition).

<img width="635" alt="screen shot 2017-10-10 at 12 12 01 pm" src="https://user-images.githubusercontent.com/3878933/31400292-af056900-adb4-11e7-82b2-03ebf39a2e57.png">
